www/js/diary/list.js

```diff
@@ -25,13 +26,17 @@ angular.module('emission.main.diary.list',['ui-leaflet',
                                     $ionicActionSheet,
                                     ionicDatePicker,
                                     leafletData, Timeline, CommonGraph, DiaryHelper,
-    Config, PostTripManualMarker, ConfirmHelper, nzTour, KVStore, Logger, UnifiedDataLoader, $ionicPopover, $ionicModal, EnketoSurveyLaunch, $translate) {
+    Config, PostTripManualMarker, ConfirmHelper, nzTour, KVStore, Logger, UnifiedDataLoader, $ionicPopover, $ionicModal, EnketoSurveyLaunch, EnketoSurvey, $translate) {
   console.log("controller DiaryListCtrl called");
     var MODE_CONFIRM_KEY = "manual/mode_confirm";
     var PURPOSE_CONFIRM_KEY = "manual/purpose_confirm";
 
     $scope.confirmSurvey = function(trip) {
-      EnketoSurveyLaunch.launch($scope, 'ConfirmSurvey', { trip: trip });
+      EnketoSurveyLaunch.launch($scope, 'ConfirmSurvey', { trip: trip }).then(function() {
+        $scope.$apply(function() {
+          $scope.updateCustomSurveyLabelForTrip(trip);
```

Comment:
The code to update the survey label as part of the survey close is added here, as part of this PR. Before this, the only time the displayed label was updated was as part of `populateSurveyAnswerFromTimeline`